tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -1461,7 +1461,7 @@ class PlaceholderExpansionScanner {
     bool walkToTypeReprPre(TypeRepr *T) override {
       if (auto *FTR = dyn_cast<FunctionTypeRepr>(T)) {
         FoundFunctionTypeRepr = true;
-        if (auto *TTR = dyn_cast_or_null<TupleTypeRepr>(FTR->getArgsTypeRepr())) {
+        if (auto *TTR = FTR->getArgsTypeRepr()) {
```

Comment:
This probably doesn't need to be an `if` at all now.

Reply:
It's the `_or_null` that's got me scared.

Reply:
I'm not sure how you'd ever get a FunctionTypeRepr with *no* argument types, but fair enough.

Reply:
It was probably defensive coding. If the tests pass without it I would just remove it.